### ISSUE TITLE
How to configure the KV Cache Manager for SSD?

### ISSUE THREAD

**Issue by TayQiXiang** [ai-dynamo/dynamo#1346]
### Feature request

I'm wondering how we can set how much KV cache should be offloaded to SSD and how can we select which SSD mount point to offload?

### Describe the problem you're encountering

Right now documentation does not have a clear way of enabling SSD offload for KV cache.

**Comment by vyalamar** [comment]
How to self assign this ? Can I take this.

**Comment by jthomson04** [comment]
We are actively working on integrating KVBM into vLLM. Until then, we don't yet support SSD offloading. However, you can still do host offloading through the pre-existing features in vLLM, TRT-LLM, and others.

**Comment by TayQiXiang** [comment]
Is there a rough time frame as it to when will it be completed and PRed in? It is quite critical to evaluate the performance of the KVBM and in extension, dynamo.

**Comment by TayQiXiang** [comment]
> We are actively working on integrating KVBM into vLLM. Until then, we don't yet support SSD offloading. However, you can still do host offloading through the pre-existing features in vLLM, TRT-LLM, and others.

Host offloading in this case means offload model or KV cache? Does vllm itself support kv cache offload?

**Comment by jthomson04** [comment]
Offloading in this case refers to KV cache. vLLM supports kv cache offload through their LMCache integration.

**Comment by didoteebin** [comment]
@jthomson04 --- I see  Host -> Disk offload code in offload.rs , so what do you mean by not supported right now pls?

// Host -> Disk offload
        let host_to_disk_task = OffloadManager::offload_worker(
            this.host.clone(),
            this.disk.clone(),
            host_offload_rx,
            Arc::new(TransferBatcher::new(
                DiskTransferManager::new(
                    transfer_ctx.clone(),
                    MAX_CONCURRENT_TRANSFERS,
                    &async_rt_handle,
                    cancellation_token.clone(),
                ),
                MAX_TRANSFER_BATCH_SIZE,
                &async_rt_handle,
                cancellation_token.clone(),
            )),
            cancellation_token.clone(),
        );
        CriticalTaskExecutionHandle::new_with_runtime(
            |_| host_to_disk_task,
            cancellation_token.clone(),
            "Host -> Disk offload worker",
            &async_rt_handle,
        )?
        .detach();

**Comment by jthomson04** [comment]
KVBM supports it, but there isn't yet any way to use disk offloading (or kvbm as a whole) in an inference framework. We're currently working on getting KVBM to work with vLLM; see the https://github.com/ai-dynamo/dynamo/tree/ryan/vllm-mega branch.

**Comment by sectorsize512** [comment]
Looks like some KV Cache offloading code was merged in 0.3.0 - is it working? If so, how can it be configured? TY.

**Comment by jthomson04** [comment]
Yes, KVBM 0.3.0 supports host + disk offloading. However, KVBM isn't yet integrated into any inference framework, so this offloading functionality only exists as a standalone module. We're actively working on integrating KVBM + offloading functionality into vLLM.

**Comment by jthomson04** [comment]
Closing this. Please reopen if you have any further questions.
